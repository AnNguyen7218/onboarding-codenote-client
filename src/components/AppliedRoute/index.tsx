import React from 'react';
import { Route } from 'react-router-dom';

interface Props {
  component: new() => React.Component<any, any>,
  props?: Object,
  [x: string]: any,
}

const AppliedRoute = ({ component: C, props: cProps, ...rest }: Props) =>
  <Route {...rest} render={ props => <C {...props} {...cProps} /> } />;

export default AppliedRoute;
