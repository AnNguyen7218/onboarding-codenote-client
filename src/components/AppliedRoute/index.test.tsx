import { shallow, mount } from 'enzyme';
import React from 'react';
import { Route } from 'react-router';
import AppliedRoute from './index';
import Home from '../../screens/Home/index'

test('AppliedRoute renders without crashing', () => {
  let props
  const wrapper = shallow(<AppliedRoute {...props}/>);
  expect(wrapper).toMatchSnapshot();
});
