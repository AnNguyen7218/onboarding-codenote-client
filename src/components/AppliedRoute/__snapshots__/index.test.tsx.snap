// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppliedRoute renders without crashing 1`] = `
<Route
  render={[Function]}
/>
`;
